frontend/src/concepts/roleBinding/RoleBindingPermissionsTableRow.tsx

```diff
@@ -2,57 +2,67 @@ import * as React from 'react';
 import { ActionsColumn, Tbody, Td, Tr } from '@patternfly/react-table';
 import {
   Button,
+  Icon,
   Split,
   SplitItem,
   Text,
   Timestamp,
   TimestampTooltipVariant,
+  Tooltip,
 } from '@patternfly/react-core';
-import { CheckIcon, TimesIcon } from '@patternfly/react-icons';
+import { CheckIcon, OutlinedQuestionCircleIcon, TimesIcon } from '@patternfly/react-icons';
 import { RoleBindingKind } from '~/k8sTypes';
 import { relativeTime } from '~/utilities/time';
-import { castProjectSharingRoleType, firstSubject, roleLabel } from './utils';
-import { ProjectSharingRBType, ProjectSharingRoleType } from './types';
-import ProjectSharingNameInput from './ProjectSharingNameInput';
-import ProjectSharingPermissionSelection from './ProjectSharingPermissionSelection';
+import { castRoleBindingPermissionsRoleType, firstSubject, roleLabel } from './utils';
+import { RoleBindingPermissionsRBType, RoleBindingPermissionsRoleType } from './types';
+import RoleBindingPermissionsNameInput from './RoleBindingPermissionsNameInput';
+import RoleBindingPermissionsPermissionSelection from './RoleBindingPermissionsPermissionSelection';
 
-type ProjectSharingTableRowProps = {
+type RoleBindingPermissionsTableRowProps = {
   obj: RoleBindingKind;
-  type: ProjectSharingRBType;
+  type: RoleBindingPermissionsRBType;
   isEditing: boolean;
+  defaultGroupName?: string;
+  permissionOptions: {
+    type: RoleBindingPermissionsRoleType;
+    description: string;
+  }[];
   typeAhead?: string[];
-  onChange: (name: string, roleType: ProjectSharingRoleType) => void;
+  onChange: (name: string, roleType: RoleBindingPermissionsRoleType) => void;
   onCancel: () => void;
   onEdit: () => void;
   onDelete: () => void;
 };
 
 const defaultValueName = (obj: RoleBindingKind) => firstSubject(obj);
-const defaultValueRole = (obj: RoleBindingKind) => castProjectSharingRoleType(obj.roleRef.name);
+const defaultValueRole = (obj: RoleBindingKind) =>
+  castRoleBindingPermissionsRoleType(obj.roleRef.name);
 
-const ProjectSharingTableRow: React.FC<ProjectSharingTableRowProps> = ({
+const RoleBindingPermissionsTableRow: React.FC<RoleBindingPermissionsTableRowProps> = ({
   obj,
   type,
   isEditing,
+  defaultGroupName,
+  permissionOptions,
   typeAhead,
   onChange,
   onCancel,
   onEdit,
   onDelete,
 }) => {
   const [roleBindingName, setRoleBindingName] = React.useState(defaultValueName(obj));
-  const [roleBindingRoleRef, setRoleBindingRoleRef] = React.useState<ProjectSharingRoleType>(
-    defaultValueRole(obj),
-  );
+  const [roleBindingRoleRef, setRoleBindingRoleRef] =
+    React.useState<RoleBindingPermissionsRoleType>(defaultValueRole(obj));
   const [isLoading, setIsLoading] = React.useState(false);
   const createdDate = new Date(obj.metadata.creationTimestamp || '');
+  const isDefaultGroup = obj.metadata.name === defaultGroupName;
```

Comment:
Ok so don't kill me, but technically `obj.metadata.name` is the rolebinding name, not the group name (the rolebinding's subject). The logic I was imagining is that `isDefaultGroup` is true if the `obj.subjects` contains a subject of kind `'Group'` and name equal to `defaultGroupName`. Buuut you could also just rename this prop to `defaultRoleBindingName`.

Reply:
🔪